components/layout/inline.rs

```diff
@@ -684,16 +684,14 @@ impl LineBreaker {
         }
 
         // Determine if an ellipsis will be necessary to account for `text-overflow`.
-        let mut need_ellipsis = false;
         let available_inline_size = self.pending_line.green_zone.inline -
             self.pending_line.bounds.size.inline - indentation;
-        match (fragment.style().get_text().text_overflow,
-               fragment.style().get_box().overflow_x) {
-            (text_overflow::T::clip, _) | (_, overflow_x::T::visible) => {}
-            (text_overflow::T::ellipsis, _) => {
-                need_ellipsis = fragment.margin_box_inline_size() > available_inline_size;
-            }
-        }
+
+        let need_ellipsis = match fragment.style().get_text().text_overflow.first {
+            longhands::text_overflow::Side::Clip => false,
+            longhands::text_overflow::Side::Ellipsis |longhands::text_overflow::Side::String(..) =>
```

Comment:
Any reason why we're using need_ellipsis and an unwrap/unreachable instead of creating an Option here and restructuring it and passing it down to transform? IIRC this was the case in the PR originally, but it changed.